src/companion.jl

```diff
@@ -15,86 +16,78 @@ Also, directly from a polynomial:
 ```julia
 julia> using Polynomials
 
-julia> P=Polynomial([2.0,3,4,5])
+julia> P = Polynomial(2:5)
 Polynomial(2 + 3x + 4x^2 + 5x^3)
 
-julia> C=Companion(P)
+julia> C = Companion(P)
 3×3 Companion{Float64}:
  0.0  0.0  -0.4
  1.0  0.0  -0.6
  0.0  1.0  -0.8
 ```
 """
-struct Companion{T} <: AbstractArray{T, 2}
+struct Companion{T} <: AbstractMatrix{T}
     c :: Vector{T}
 end
 
-# Generate companion matrix from a polynomial
+Companion(v::AbstractVector) = Companion(collect(v))
```

Comment:
Why is this new constructor necessary? Julia will automatically perform `convert(Vector{T}, v)` for assignment to the field.

Reply:
Hmm, what version of Julia are you using that does that automatic conversion?  I am using 1.8.2, and if I remove this constructor then it throws an error:
```julia
julia> Companion(1:3)
ERROR: MethodError: no method matching Companion(::UnitRange{Int64})
Closest candidates are:
  Companion(::Polynomial) at ~/.julia/dev/SpecialMatrices/src/companion.jl:37
  Companion(::Vector{T}) where T at ~/.julia/dev/SpecialMatrices/src/companion.jl:30
```
I'm pretty sure we need this.

Reply:
I see, it's due to the parameter.

```julia
julia> Companion{Int}(1:3)
3×3 Companion{Int64}:
 0  0  -1
 1  0  -2
 0  1  -3
```